src/translate.rs

```diff
@@ -1088,26 +1089,25 @@ impl FromGlibPtrNone<*const c_char> for String {
 impl FromGlibPtrFull<*const c_char> for String {
     #[inline]
     unsafe fn from_glib_full(ptr: *const c_char) -> Self {
-        let res = from_glib_none(ptr);
-        glib_ffi::g_free(ptr as *mut _);
-        res
+        let holder = cstringholder::CStringHolder::new(ptr);
+        String::from(holder)
     }
 }
 
 impl FromGlibPtrNone<*mut c_char> for String {
     #[inline]
     unsafe fn from_glib_none(ptr: *mut c_char) -> Self {
         assert!(!ptr.is_null());
-        String::from_utf8_lossy(CStr::from_ptr(ptr).to_bytes()).into_owned()
+        let holder = cstringholder::CStringHolder::new(ptr as *const c_char);
+        String::from(holder)
```

Comment:
This is creating a copy. What is needed is that functions that return an owned string get the new type as return type

Reply:
Yeah, I forgot to regenerate the bindings :)